src/com/cft/contactmerge/StreetAddressMatchLogic.java

```diff
@@ -2,13 +2,185 @@
+        put("st", "street");
+        put("street", "street");
+        put("ave", "avenue");
+        put("avenue", "avenue");
+        put("dr", "drive");
+        put("drive", "drive");
+        put("ln", "lane");
+        put("lane", "lane");
+        put("trl", "trail");
+        put("trail", "trail");
+        put("cir", "circle");
+        put("circle", "circle");
+        put("blvd", "boulevard");
+        put("boulevard", "boulevard");
+        put("rd", "road");
+        put("road", "road");
+    }};
+
+    private static final Hashtable<String,String> compassDirection = new Hashtable<String,String>() {{
+        put("n", "north");
+        put("north", "north");
+        put("s", "south");
+        put("south", "south");
+        put("e", "east");
+        put("east", "east");
+        put("w", "west");
+        put("west", "west");
+    }};
+
+    /*******************************************************************************************************************
+     *************************************************** Get Methods ***************************************************
+     *******************************************************************************************************************/
+    public static Hashtable<String,String> getRoadTypes(){
+        return roadTypes;
+    }
+
+    public static Hashtable<String,String> getCompassDirection(){
+        return compassDirection;
+    }
+
+    public static ArrayList<String> getPoBoxTypes(){
+        return poBoxTypes;
+    }
+
     /*******************************************************************************************************************
      ************************************************* Matching Methods ************************************************
      *******************************************************************************************************************/
-    public static AnswerType getStreetAddressMatchResult(ArrayList<String> stAddress1, ArrayList<String> stAddress2){
-        // ToDo: Fill in code to obtain matching result
+    public static AnswerType getStreetAddressMatchResult(ArrayList<String> stAddressParts1, ArrayList<String> stAddressParts2){
+        // Precondition: All addressParts must follow the rules from the setNormalizeStreetAddress()
+        String st1Direction = "";
+        String st2Direction = "";
+        String st1RoadType = "";
+        String st2RoadType = "";
+
+        // get return on po box addressses (if either one is a po box)
+        if (isAddressPoBox(stAddressParts1) && isAddressPoBox(stAddressParts2)){
+            // Strip all characters out of address1, and only leave numbers
+            for (int part = 0; part < stAddressParts1.size(); part++){
+                if (!isInteger(stAddressParts1.get(part))){
+                    stAddressParts1.remove(part);
+                    part--;
+                }
+            }
+            // Strip all characters out of address2, and only leave numbers
```

Comment:
Duplicate code... Should move into a separate helper function. Added benefit is that you can directly test the routine and make sure you hit all edge cases.

Reply:
I thought about this... but it's doing two things...
1. seeing if the address has a direction, and storing that direction for comparison later....
2. removing the direction from the address array.

I couldn't think of a way two return both of these at the same time. The only option I could think of was having two separate methods for each of these... but that would mean 2 extra for loops (4 total). Maybe we can discuss this Monday.